core/network/src/discovery.rs

```diff
@@ -26,6 +26,38 @@ use std::{cmp, num::NonZeroU8, time::Duration};
 use tokio_io::{AsyncRead, AsyncWrite};
 use tokio_timer::{Delay, clock::Clock};
 
+//! Discovery mechanisms of Substrate.
+//!
+//! The `DiscoveryBehaviour` struct implements the `NetworkBehaviour` trait of libp2p and is
+//! responsible for discovering the nodes that are part of the network.
```

Comment:
```suggestion
//! responsible for discovering other nodes that are part of the network.
```
Just to not allow the confusion that _the nodes_ can be read as _all nodes_. Apart from very small networks, a node is very unlikely to ever discover all other nodes, especially not if "discovered" actually means "is in my DHT routing table", which is how discovered is defined by the implementation of `DiscoveryBehaviour`.